### PR TITLE
Update referrers response

### LINKED ISSUE #1410: [Feat]: referrers API MUST respond with the response header `OCI-Subject: true`
https://github.com/opencontainers/distribution-spec/commit/16cc054de70442ea08e1aa269cee1c6cda5e7c96

```
When processing a request for an image or artifact manifiest with the `subject` field, a registry implementation that supports the [referrers API](#listing-referrers) MUST respond with the response header `OCI-Subject-Processed: true` to indicate to the client that the registry understands the request's `subject`.
```

### PR DESCRIPTION
**Pull request by laurentiuNiculae**
**What type of PR is this?**

**Which issue does this PR fix**:
Closes #1410

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/local/local.go [hunk 1]
```diff
@@ -499,11 +499,11 @@ func (is *ImageStoreLocal) PutImageManifest(repo, reference, mediaType string, /
 
 	updateIndex, oldDgst, err := storage.CheckIfIndexNeedsUpdate(&index, &desc, is.log)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	if !updateIndex {
-		return desc.Digest, nil
+		return desc.Digest, "", nil
```
Comment (andaaron) [inline pkg/storage/local/local.go:506]: Shouldn't the subject digest be returned also in this case?

File: pkg/storage/s3/s3.go [hunk 2]
```diff
@@ -404,11 +404,11 @@ func (is *ObjectStorage) PutImageManifest(repo, reference, mediaType string, //n
 
 	updateIndex, oldDgst, err := storage.CheckIfIndexNeedsUpdate(&index, &desc, is.log)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	if !updateIndex {
-		return desc.Digest, nil
+		return desc.Digest, "", nil
```
Comment (andaaron) [inline pkg/storage/s3/s3.go:411]: Shouldn't this also return the subject digest?

**Review by andaaron — APPROVED** [review]

File: pkg/api/routes.go [hunk 3]
```diff
@@ -656,6 +656,10 @@ func (rh *RouteHandler) UpdateManifest(response http.ResponseWriter, request *ht
 		}
 	}
 
+	if subjectDigest.String() != "" {
+		response.Header().Set(constants.SubjectDigestKey, subjectDigest.String())
```
Comment (rchincha) [inline pkg/api/routes.go:660]: Where is the "OCI-Subject-Processed" header set?

Comment (laurentiuNiculae) [reply pkg/api/routes.go:660]: In the latest version of the [distribution spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pushing-manifests-with-subject) there is no "OCI-Subject-Processed" header. It was changed to "OCI-Subject".

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**